Userland/Libraries/LibWeb/URL/URLSearchParams.cpp

```diff
@@ -156,39 +160,49 @@ size_t URLSearchParams::size() const
     return m_list.size();
 }
 
-void URLSearchParams::append(DeprecatedString const& name, DeprecatedString const& value)
+WebIDL::ExceptionOr<void> URLSearchParams::append(String const& name, String const& value)
 {
+    auto& vm = realm().vm();
+
     // 1. Append a new name-value pair whose name is name and value is value, to list.
-    m_list.empend(name, value);
+    TRY_OR_THROW_OOM(vm, m_list.try_empend(name, value));
     // 2. Update this.
-    update();
+    TRY(update());
+
+    return {};
 }
 
-void URLSearchParams::update()
+JS::ThrowCompletionOr<void> URLSearchParams::update()
```

Comment:
Hmm, why do we need a `ThrowCompletionOr` here and can't just propagate the `ExceptionOr` from `to_string()`?

Reply:
Uh, I am not really sure, maybe trying to be clever? Anyway changed it to `WebIDL::ExceptionOr<>`